src/psyclone/tests/psyir/symbols/data_type_symbol_test.py

```diff
@@ -72,3 +72,22 @@ def test_datatypesymbol_copy():
     assert isinstance(new_symbol.datatype, DeferredType)
     assert new_symbol.visibility == Symbol.Visibility.PRIVATE
     assert isinstance(new_symbol.interface, UnresolvedInterface)
+
+
+def test_data_type_symbol_copy_properties():
+    ''' Check that the copy_properties() method works as expected. '''
+    symbol = DataTypeSymbol("origin", ArrayType(REAL_SINGLE_TYPE, [1, 2]))
+    new_sym = DataTypeSymbol("new_name", DeferredType())
+
+    new_sym.copy_properties(symbol)
+
+    # new_sym name should be unchanged, but its datatype should be updated
+    assert new_sym.name == "new_name"
+    assert new_sym.datatype == symbol.datatype
```

Comment:
As commented in the implementation, I'm concerned that the `datatype` of `new_sym` is now a reference to the `datatype` of `symbol`?

Reply:
Yes, but this is similar to what we do with `== INTEGER_TYPE`, so if it is not what we want/need this needs a redesign expanding outside the scope of this PR.